lib/service/service.go

```diff
@@ -3215,11 +3217,15 @@ func (process *TeleportProcess) initSSH() error {
 			}
 
 			go s.Serve(listener)
-		} else {
-			// Start the SSH server. This kicks off updating labels and starting the
-			// heartbeat.
-			if err := s.Start(); err != nil {
-				return trace.Wrap(err)
+		}
+
+		if conn.UseTunnel() {
+			if !useLocalListener {
+				// Start the SSH server. This kicks off updating labels and starting the
+				// heartbeat.
+				if err := s.Start(); err != nil {
```

Comment:
nit: this conditional start reads a bit funny when scanning this logic.  Might be worth an explanatory comment.  Ex:

```golang
// if local listener setup wasn't run, the server still needs to be started
```

Reply:
I'd consider making it even more explicit in the code and setting a flag for `serverStarted` to true right before the `go s.Serve(listener)` and checking for `conn.UseTunnel() && !serverStarted` here.